### PR TITLE
[SPARK-25665][SQL][TEST] Refactor ObjectHashAggregateExecBenchmark to…

### PR DESCRIPTION
**Pull request by peter-toth**
## What changes were proposed in this pull request?

Refactor ObjectHashAggregateExecBenchmark to use main method

## How was this patch tested?

Manually tested:
```
bin/spark-submit --class org.apache.spark.sql.execution.benchmark.ObjectHashAggregateExecBenchmark --jars sql/catalyst/target/spark-catalyst_2.11-3.0.0-SNAPSHOT-tests.jar,core/target/spark-core_2.11-3.0.0-SNAPSHOT-tests.jar,sql/hive/target/spark-hive_2.11-3.0.0-SNAPSHOT.jar --packages org.spark-project.hive:hive-exec:1.2.1.spark2 sql/hive/target/spark-hive_2.11-3.0.0-SNAPSHOT-tests.jar
```
Generated results with:
```
SPARK_GENERATE_BENCHMARK_FILES=1 build/sbt "hive/test:runMain org.apache.spark.sql.execution.benchmark.ObjectHashAggregateExecBenchmark"
```

### CODE REVIEW

File: sql/hive/src/test/scala/org/apache/spark/sql/execution/benchmark/ObjectHashAggregateExecBenchmark.scala [hunk 1]
```diff
@@ -21,207 +21,212 @@ import scala.concurrent.duration._
...
-      sparkSession.conf.set(SQLConf.USE_OBJECT_HASH_AGG.key, "true")
-      df.groupBy($"id" / (N / 4) cast LongType).agg(percentile_approx($"id", 0.5)).collect()
-    }
-
-    benchmark.addCase("object agg w/ group by w/ fallback") { _ =>
-      sparkSession.conf.set(SQLConf.USE_OBJECT_HASH_AGG.key, "true")
-      sparkSession.conf.set(SQLConf.OBJECT_AGG_SORT_BASED_FALLBACK_THRESHOLD.key, "2")
-      df.groupBy($"id" / (N / 4) cast LongType).agg(percentile_approx($"id", 0.5)).collect()
+/**
+ * Benchmark to measure read performance with Filter pushdown.
```
Comment (wangyum) [inline sql/hive/src/test/scala/org/apache/spark/sql/execution/benchmark/ObjectHashAggregateExecBenchmark.scala]: read performance with Filter pushdown?

Comment (peter-toth) [reply sql/hive/src/test/scala/org/apache/spark/sql/execution/benchmark/ObjectHashAggregateExecBenchmark.scala]: Thanks @wangyum , fixed.

**Comment by dongjoon-hyun** [comment]
ok to test

File: sql/hive/src/test/scala/org/apache/spark/sql/execution/benchmark/ObjectHashAggregateExecBenchmark.scala [hunk 2]
```diff
@@ -21,207 +21,212 @@ import scala.concurrent.duration._
...
+ *   3. generate result: SPARK_GENERATE_BENCHMARK_FILES=1 build/sbt "hive/test:runMain <this class>"
+ *      Results will be written to "benchmarks/ObjectHashAggregateExecBenchmark-results.txt".
+ * }}}
+ */
+object ObjectHashAggregateExecBenchmark extends BenchmarkBase with SQLHelper {
+
+  val spark: SparkSession = TestHive.sparkSession
+
+  override def runBenchmarkSuite(): Unit = {
+    runBenchmark("Hive UDAF vs Spark AF") {
```
Comment (dongjoon-hyun) [inline sql/hive/src/test/scala/org/apache/spark/sql/execution/benchmark/ObjectHashAggregateExecBenchmark.scala]: Hi, @peter-toth . Thank you for making this PR.
Currently, `runBenchmarkSuite` is too long. Could you make a separate function for each test case? For example, `ignore("Hive UDAF vs Spark AF")` can be a single function. And `runBenchmarkSuite` will call a series of those functions.

**Review by dongjoon-hyun — COMMENTED** [review]
Hi, @peter-toth .
Thank you for updating. I made a [PR](https://github.com/peter-toth/spark/pull/2) to you. Could you review and merge that?

**Comment by peter-toth** [comment]
Thanks @dongjoon-hyun for the fixes. Merged.

**Review by dongjoon-hyun — APPROVED** [review]
Thank you, @peter-toth and @wangyum .
+1, LGTM.

Merged to master.

**Comment by peter-toth** [comment]
Thanks @dongjoon-hyun , @wangyum for the review.

**Pull request closed without merging**